[Economy] Add embed to leaderboard

TurnrDev opened the pull request:
# Enhancement request

#### Describe the enhancement

add embed support to leaderboard as requested on #4104

#### Does this enhancement break existing functionality?

Yes

Closes #4104

Commit:
Bank Leaderboard embed

Commit:
Add embed formatting

Commit:
Support sometimes not sending an embed & more consistent output

TurnrDev commented:
Embed: ![image](https://user-images.githubusercontent.com/11667059/90322187-06540e00-df49-11ea-89f4-4afb7f1ae633.png)

The embed author is set to the guild name and icon, unless global, when it's set to the bots name and icon

No embed: 
![image](https://user-images.githubusercontent.com/11667059/90322360-27b5f980-df4b-11ea-8d95-807270e9a75b.png)

Drapersniper commented on redbot/cogs/economy/economy.py:
```diff
@@ -536,15 +543,58 @@ async def leaderboard(self, ctx: commands.Context, top: int = 10, show_global: b
+                            "You are {author_pos}/{len_board}. Page {page_num}/{page_len}."
+                            " Use {left} and {right} to navigate."
+                        ).format(
+                            author_pos=bank.get_leaderboard_position(author),
```
this is missing an await on `bank.get_leaderboard_position` however please dont do this, the this to a variable outside the loop and then reference this variable here and in L578

TurnrDev replied:
Thank you for spotting that! I've now fixed this in 516cf91fc6310968848f431410764debbe4d828d :)

Drapersniper reviewed (changes requested):
Please do the same thing and assign variables rather than making a list and slicing it on like 558,559, 582 and 583

Line 552 and 576 will tell the users what reaction to use to move, but it will show even if there's only 1 page, I would reccomend this part is not added to the the embed at all

Commit:
Fixed author position await issue

Mentioned by Drapersniper here -> https://github.com/Cog-Creators/Red-DiscordBot/pull/4184/files/27a8ff3297955b192744e578f323b622061c8c21#r471098141

Commit:
Move embed_requested variable one up as it'll annoy me

Commit:
The footer message is assigned to a variable now,

and we scrapped the instructions on how to navigate the pages. Instructions aren't added anywhere else on the bot and should be the scope of a separate PR.

TurnrDev commented:
> Please do the same thing and assign variables rather than making a list and slicing it on like 558,559, 582 and 583
> 
> Line 552 and 576 will tell the users what reaction to use to move, but it will show even if there's only 1 page, I would reccomend this part is not added to the the embed at all

16a4b7e0331fe0143428c8ce3265b9f38fb68c23
Made changes :) 

Removed instructions as instructions aren't added anywhere else on the bot and should be the scope of a separate PR.

Drapersniper reviewed (approved):
Awesome its looking smexy... thanks for this.
![image](https://user-images.githubusercontent.com/27962761/90497583-251bf580-e13f-11ea-91e5-2c6328a3afea.png)


Once I'll further tweak the UI in #4062 as part of the dpy implementation, but stils already looking pretty darn good!

Drapersniper reviewed (changes requested):
so turns out ad i looked at the screenshot a second time ... the Position is off `25/10`

Commit:
Removed get_leaderboard_position due inconsistencies becoming apparent

Drapersniper reviewed (approved):
Thansk for the PR

TurnrDev commented:
No no, thank you for the challenge @Drapersniper :)
